src/_shared/components/Toast/ToastList.tsx

```diff
@@ -0,0 +1,38 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import ReactDOM from 'react-dom';
+
+import { Toast } from './Toast';
+import { ToastData } from './toast.d';
+
+interface Props {
+  toastList: ToastData[];
+}
+
+export const ToastList = ({ toastList }: Props) => {
+  const [element, setElement] = useState<HTMLElement | null>(null);
+  const ref = useRef(null);
+
+  useEffect(() => {
+    setElement(document.getElementById('toast-root'));
+  }, []);
+
+  if (!element) return <></>;
+
+  return ReactDOM.createPortal(
+    <ul
+      className='fixed top-0 right-0 flex flex-col-reverse w-fit h-fit max-h-full'
+      ref={ref}
+    >
+      {toastList.map((toast) => (
```

Comment:
```suggestion
      {toastList.map(({id, type, text}) => (
```
구조분해 가능한거로 알고있는데 요렇게 하면 좀 더 item쪽 코드가 깔끔해질거 같아요.